url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/templating/TemplateReader.java

```diff
@@ -0,0 +1,84 @@
+package stv6.templating;
+
+import java.io.BufferedReader;
+import java.io.EOFException;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class TemplateReader {
+    private File sourceFile;
+	private BufferedReader file;
+	private StringBuilder lastLine = null;
+	private boolean useLast = false;
+	
+	/** 
+	 * For extending this class to not use the filesystem
+	 */
+	protected TemplateReader() {}
+	
+	public TemplateReader(File source) throws FileNotFoundException {
+	    sourceFile = source;
+		file = new BufferedReader(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *file*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).